metadata/write_jsons.py

```diff
@@ -0,0 +1,256 @@
+    :return: all_sims_dict or None
+    """
+    # Get all ch_log directories
+    hh = get_hh(fault_dir)
+    result_dict = {}
+
+    # Find all ch_log and Rlog directories (to allow handling of the different
+    # folder structures
+    ch_log_dirs = glob.glob(os.path.join(fault_dir, "**", "ch_log"))
+    rlog_dirs = glob.glob(os.path.join(fault_dir, "**", "LF", "**", "Rlog"),
+                          recursive=True)
+    if len(ch_log_dirs) == 0:
+        print("No ch_log directories found for {}".format(fault_dir))
+        return
+
+    for ch_log_dir in ch_log_dirs:
+        # Iterate over the log files
+        for f in os.listdir(ch_log_dir):
+            segs = f.split('.')
+            sim_type = segs[0]
+            # exclude submit time log file (have 3 segs)
+            if len(segs) == 4 and sim_type in KEY_DICT.keys():
+                with open(os.path.join(ch_log_dir, f), 'r') as log_file:
+                    buf = log_file.readlines()
+
+                raw_data = buf[0].strip().split()
+                realization = raw_data[0]
+                data = raw_data[2:]
+
+                # Realization entry already exists
+                if realization in result_dict.keys():
+                    result_dict[realization].update({sim_type: {}})
+                # Create the realization entry
+                else:
+                    result_dict[realization] = {'common': {'hh': hh},
+                                                  sim_type: {}}
+
+                # Retrieve the metadata
+                for i in range(len(data)):
+                    key = KEY_DICT[sim_type][i]
+                    if key == 'run_time':
+                        data[i] = "{:.5f} hour".format(float(data[i]))
+                    result_dict[realization][sim_type][key] = data[i]
+
+                # Extra LF metadata
+                if sim_type == 'LF' and \
+                        result_dict[realization][sim_type].get('total_memory_usage') is None:
+                    # Get the rlog dir for the current realization
+                    cur_rlog_dir = [rlog_dir for rlog_dir in rlog_dirs if
+                                    realization in rlog_dir.split("/")][0]
+
+                    result_dict[realization][sim_type]['total_memory_usage'] = \
+                        get_one_realization_memo_cores(cur_rlog_dir)
+
+                result_dict[realization][sim_type][
+                    'submit_time'] = get_submit_time(ch_log_dir, sim_type,
+                                                     realization)
+    return result_dict
+
+
+def get_one_realization_memo_cores(realization_rlog_dir):
+    """
+    :param realization_rlog_dir: **/LF/**/Rlog
+    :return:total_memory usage for each realization in GB
+    """
+    os.chdir(realization_rlog_dir)
+    cmd = "grep 'total for model' *.rlog"
+    output = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True).communicate()[0].decode()
+    total = 0
+    unit = ''
+
```

Comment:
I sometimes find it helpful to show output example before playing with strip() and split() - it makes the future debugging extremely easy (when *.rlog changes its format and the output suddenly looks different, we can immediately know what's going on). You don't need to change anything - just my opinion...

Reply:
I didn't actually write that particular function, I think the reason why github shows the whole file as new is because I moved the file, which is a bit annoying. I did modify this file heavily as well though. 

Yeah I agree, I also don't think this function should be using an external cmd, this should be done in python in my opinion. I have been modifying this script quite heavily recently, so I'll change that next time I'm working on it (the most up to date version of this file is in a different branch).